schema_utils.py

```diff
@@ -627,3 +627,27 @@ def is_valid_audio_language_code(obj):
             bool. Whether the given object is a valid audio language code.
         """
         return utils.is_valid_language_code(obj)
+
+    @staticmethod
+    def is_valid_thread_id(obj):
+        # type: (Text) -> bool
+        """Checks if the given obj (a string) represents a valid thread id.
+        The id has the form [entity_type].[entity_id].[generated_string]
+
+        Args:
+            obj: str. The thread id to verify.
+
+        Returns:
+            bool. Whether the given object is a valid thread id.
+        """
+        thread_id_must_starts_with = [
```

Comment:
@vojtechjelinek I have looked into constants.ts, feconf.py and concluded the following points.
1. [Here](https://github.com/oppia/oppia/blob/6cc7d35755980e7cc67fe8d1cd1403357e94267a/feconf.py#L1436) a list is maintained for suggestion target type choices.
2. And [here](https://github.com/oppia/oppia/blob/6cc7d35755980e7cc67fe8d1cd1403357e94267a/feconf.py#L170) all of the entity_type are written.
3. And finally [here](https://github.com/oppia/oppia/blob/6cc7d35755980e7cc67fe8d1cd1403357e94267a/assets/constants.ts#L40) in constants.ts another few activity_types are written.

I am not sure for thread_id which of the entity_type to use.
kindly help me regarding this.
Thank you.

Reply:
Not sure, from where did you create the list that you have rn?

Reply:
I have created this list from constants.ts. I also wanted to ask that if the above solution is accepted i.e., to create a general validator to check regex then maybe this list will not be required.
Please confirm this, thank you.

Reply:
Which constants exactly?

Reply:
From these lines: https://github.com/oppia/oppia/blob/6cc7d35755980e7cc67fe8d1cd1403357e94267a/assets/constants.ts#L40
Thank you.